becquerel/core/fitting.py

```diff
@@ -604,13 +618,17 @@ def fit(self, backend="lmfit"):
         y_roi_norm = self.y_roi
         if self.dx is not None:
             y_roi_norm = self.y_roi / self.dx_roi
-        if backend.lower().strip() == "lmfit":
+
+        self.backend = backend.lower().strip()
+
+        if self.backend == "lmfit":
             # Perform the fit, weighted by 1/uncertainties.
             weights = self.y_unc_roi ** -1.0
```

Comment:
@cosama aren't the weights normally 1/uncertainties^2 ? Or is this an `lmfit` convention?

Reply:
Good question. Would have too think about that. Either way ` 1 / self.y_unc_roi ** -1.0` should be faster.

Reply:
`lmfit` seems to define weights as inverse of standard deviation. From their [documentation](https://lmfit.github.io/lmfit-py/model.html):

```
weights (array_like, optional) – Weights to multiply (data-model) for fit residual.
```

So the weights have to have units of "inverse data"; in other words, least squares will minimize `sum(weights**2 * (data - model)**2`.

Another way to see confirm this is that `ModelResult.plot` will plot errorbars in `y` if `yerr` is given, but if not:

```
If yerr is not specified and the fit includes weights, yerr set to 1/self.weights.
```

Reply:
I suspect that under the hood the fitter would then use weights^2, but that's not an issue for here.